CMakeLists.txt

```diff
@@ -7,32 +7,36 @@ include(GNUInstallDirs)
 include(CTest)
 
 
-cmake_dependent_option(USE_VERSIONED_LIBDIR
+cmake_dependent_option(PUGIXML_USE_VERSIONED_LIBDIR
   "Use a private subdirectory to install the headers and libraries" OFF
   "CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR" OFF)
 
-cmake_dependent_option(USE_POSTFIX
+cmake_dependent_option(PUGIXML_USE_POSTFIX
   "Use separate postfix for each configuration to make sure you can install multiple build outputs" OFF
   "CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR" OFF)
 
-cmake_dependent_option(STATIC_CRT
+cmake_dependent_option(PUGIXML_STATIC_CRT
   "Use static MSVC RT libraries" OFF
   "MSVC" OFF)
 
-cmake_dependent_option(BUILD_TESTS
+cmake_dependent_option(PUGIXML_BUILD_TESTS
   "Build pugixml tests" OFF
   "BUILD_TESTING;CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR" OFF)
 
-option(BUILD_SHARED_AND_STATIC_LIBS "Build both shared and static libraries" OFF)
- # Technically not needed for this file. This is builtin.
-option(BUILD_SHARED_LIBS "Build shared instead of static library" OFF) 
+# Custom build defines
+set(PUGIXML_BUILD_DEFINES CACHE STRING "Build defines")
 
-set(BUILD_DEFINES CACHE STRING "Build defines")
+option(PUGIXML_BUILD_SHARED_AND_STATIC_LIBS "Build both shared and static libraries" OFF)
+ 
+ # Expose option to build PUGIXML as static even when the global BUILD_SHARED_LIBS variable is set
+cmake_dependent_option(PUGIXML_BUILD_SHARED_LIBS 
```

Comment:
Why not always provide this option via `option`? Or is the idea that BUILD_SHARED_LIBS overrides this one? Maybe it would make sense to default PUGIXML_BUILD_SHARED_LIBS to the value of the global BUILD_SHARED_LIBS, but still let override in either direction?

Reply:
Yes, this looks strange. I will change it to your suggestion.

Reply:
I agree that this looks strange. However, introducing a local flag `PUGIXML_BUILD_SHARED_LIBS` which defaults to whatever `BUILD_SHARED_LIBS` is set to might be confusing, since this allows you to set the global flag to `TRUE` while the local flag is `FALSE`. Any change to the global flag won't propagate to the local flag, unless you clear the CMake cache. Thus making the behavior a bit unpredictable for the users.

My vote is to remove the statement completely. This will force the users to either use the global flag, OR building the library with the `PUGIXML_BUILD_SHARED_AND_STATIC_LIBS` flag set.

Reply:
Another option would be to revert to something similar to the previous version, i.e. exposing the global option (but defaulting to the value from the cache):
`option(BUILD_SHARED_LIBS "Build shared instead of static library" BUILD_SHARED_LIBS)`

However, this might be considered bad form by the CMake purists.

Reply:
Hmm. If we completely remove this, does this mean `BUILD_SHARED_LIBS` will still work externally since it's a built-in?

Reply:
I reverted my stance on this and re-introduced the exposure of the global option `BUILD_SHARED_LIBS`. However I put the option `PUGIXML_BUILD_SHARED_AND_STATIC_LIBS` as a dependent option to this. This required me to change the build conditionals a bit to prevent confusing option clashes.

So the commit 7b1b7ee16d86acdb5207966401dccbb28b849475 will allow for three different cases:

1. Static libraries (global `BUILD_SHARED_LIBS` set to `OFF` - *The default behavior*)
2. Shared libraries (global `BUILD_SHARED_LIBS` set to `ON`)
3. Both shared and static (global `BUILD_SHARED_LIBS` set to `ON` and `PUGIXML_BUILD_SHARED_AND_STATIC_LIBS` set to `ON`)